src/watchFiles.ts

```diff
@@ -89,4 +89,8 @@ async function moveFiles(filePath: string) {
   }
 }
 
-chokidar.watch(folderToWatch, { awaitWriteFinish: true, ignorePermissionErrors: true }).on('add', async (strPath) => await moveFiles(strPath));
+if (require.main === module) {
```

Comment:
This is bad practice.

Put this command in it's own file (like `runner.ts`) to seperate your runner logic to your function logic, then you can test your functions without actually running a watcher.

Reply:
How do I prevent it from being executed? This is what I have so far:

watcher.ts: 

`export default function watch(folderToWatch: string, callback: Callback) {
    chokidar.watch(folderToWatch, { awaitWriteFinish: true, ignorePermissionErrors: true }).on('add', async (strPath) => await callback(strPath));
}`

watchFiles.ts:

To call it:
`watch(folderToWatch, moveFiles)`

Reply:
You don't import `watch.ts` in `watchFiles.ts`, you import `watchFiles.ts` from `watch.ts`

```
// watch.ts
import { moveFiles } from './watchFiles.ts';

chokidar.watch(folderToWatch, { awaitWriteFinish: true, ignorePermissionErrors: true }).on('add', async (strPath) => await moveFiles(strPath)); }
```

```
// watchFiles.ts
export moveFiles;
```

Then to run your script you do `node dist/watch.js`